pac4j-core/src/main/java/org/pac4j/core/profile/ProfileManager.java

```diff
@@ -15,44 +15,71 @@
  */
 package org.pac4j.core.profile;
 
-import org.pac4j.core.authorization.Authorizer;
+import org.pac4j.core.context.Pac4jConstants;
 import org.pac4j.core.context.WebContext;
 
 /**
- * <p>This abstract class is to be implemented in various implementation libraries,
- * to manage the current user profile, i.e. the one of the current authenticated user.</p>
- * <p>The constructors of the subclasses will tell which inputs are required in the specific libraries.</p>
+ * <p>This class is a generic way to manage the current user profile, i.e. the one of the current authenticated user.</p>
+ * <p>It may be partially ree-implemented for specific needs / frameworks.</p>
  *
- * 
  * @author Jerome Leleu
  * @since 1.8.0
  */
-public abstract class ProfileManager<C extends CommonProfile> {
+public class ProfileManager<U extends UserProfile> {
```

Comment:
Turn this abstract class into a real implementation, the abstraction is managed by using the web context.